### ISSUE TITLE
NSPanel closes when interacting with native context menu

### ISSUE THREAD

**Issue by doroved** [ahkohd/tauri-macos-menubar-app-example#37]
https://github.com/ahkohd/tauri-nspanel/issues/27#issuecomment-2052507975

Hi.
When we open the context menu and click on or off a menu item, NSPanel closes.
Is there any way to fix this? Thanks)

**UPD:**
After the NSPanel closes, we see the focus go away from the Arc browser and if you call the context menu in the NSPanel after that, it will not close.

https://github.com/ahkohd/tauri-nspanel/assets/30116541/3ca6b124-fc87-43ad-9c7e-6aa81d5c116e

**Comment by ahkohd** [comment]
Thanks for reporting this. I will get back to you with a fix.

**Comment by ahkohd** [comment]
A quick update: I have partially fixed the issue; however, the fix I introduced resulted in another problem.

You can check out this branch to see what I'm up to.

The quickest solution is to use a non-native custom menu.

**Comment by ahkohd** [comment]
Ayo! I have fixed all the issues.

[Please check out this branch and test thoroughly.](https://github.com/ahkohd/tauri-macos-menubar-app-example/tree/native-context-menu) Let me know if it works!

**Comment by ahkohd** [comment]
I'm afraid I haven't found any solution to that.

**Comment by ahkohd** [comment]
That's a little glitch; the user will probably never notice. Can we mark this issue as resolved by merging the PR?

**Comment by doroved** [comment]
@ahkohd Hello, I am still using v2. I noticed that v2.1 has undergone significant changes and has new features, but I am not using it yet.
In v2, I noticed that this problem disappeared, even though I didn't do anything. Did I understand correctly that you found a solution to this problem back in v2, or could tauri 2.7.0 have somehow influenced it, which is unlikely?

Here's a demo:

https://github.com/user-attachments/assets/6c13520b-8424-4c1b-8ea1-322ed4a5059c

Thank you for your plugin )
